sqlglot/dialects/athena.py

```diff
@@ -79,27 +104,40 @@ class Parser(Trino.Parser):
             TokenType.USING: lambda self: self._parse_as_command(self._prev),
         }
 
+    class _HiveGenerator(Hive.Generator):
+        def preprocess(self, expression: exp.Expression) -> exp.Expression:
+            # package any ALTER TABLE ADD actions into a Schema object
+            # so it gets generated as `ALTER TABLE .. ADD COLUMNS(...)`
+            # instead of `ALTER TABLE ... ADD COLUMN` which is invalid syntax on Athena
+            if isinstance(expression, exp.Alter) and expression.kind == "TABLE":
```

Comment:
The goal here is to correctly transform `ALTER TABLE ADD COLUMN` to `ALTER TABLE ADD COLUMNS` without the user having to do anything manually to the AST.

Is this the right place for AST manipulations that need the full context of the expression?

Reply:
this is one place, you could also do a transform on Alter

Reply:
this doesn't do too much with context, so probably better as a transform, look up TRANSFORMS

Reply:
I've moved it to a transform

Reply:
I've moved it out of a transform and just override `alter_sql()` directly on the Generator